compiler/qsc/src/interpret.rs

```diff
@@ -676,6 +676,7 @@ impl Interpreter {
         } else {
             let mut sim = CircuitBuilder::new(CircuitConfig {
                 base_profile: self.capabilities.is_empty(),
+                max_operations: CircuitConfig::DEFAULT_MAX_OPERATIONS,
```

Comment:
Does this impact resource estimation at all? Obviously when running the resource estimation it wouldn't be that useful if it had a low-ish cap on the number of operations.